R/recover.weaker.R

```diff
@@ -682,8 +682,11 @@ recover.weaker <- function(filename,
 
   aver.diff <- mean(diff(times))
   vec_delta_rt <- compute_delta_rt(times)
-  
-  sample_intensities <- unlist(dplyr::select(intensity_table, dplyr::contains(sample_name)), use.names = FALSE)
+
+  sample_intensities <- unlist(dplyr::select(
+    intensity_table %>% dplyr::rename_with(~ str_remove(., "_intensity")),
```

Comment:
Can you explain in details what is happening here? Can't honestly say I completely understand what this one-liner does in the old as well as the new code.

Reply:
Previously we were doing pattern matching rather than exact match for the column selection which was prone to selecting more columns with similar pattern e.g selecting by string 6_qc_64x_dil_milliq in columns 6_qc_64x_dil_milliq_intensity, 16_qc_64x_dil_milliq_intensity would return both. this change introduces exact match but as intensity_table contains colnames appended with _intensity so we are removing that before doing the exact match

Reply:
Oh I see, so it really worked by taking the sub-strings, crazy! 
Thanks for the explanation.